src/snowflake/snowpark/dataframe_writer.py

```diff
@@ -335,34 +335,181 @@ def copy_into_location(
-
-                Returns:
-                    A list of :class:`Row` objects containing unloading results.
-
-                Example::
-
-                    >>> # save this dataframe to a parquet file on the session stage
-                    >>> df = session.create_dataframe([["John", "Berry"], ["Rick", "Berry"], ["Anthony", "Davis"]], schema = ["FIRST_NAME", "LAST_NAME"])
-                    >>> remote_file_path = f"{session.get_session_stage()}/names.csv"
-                    >>> copy_result = df.write.csv(remote_file_path, overwrite=True)
-                    >>> copy_result[0].rows_unloaded
-                    3
-                """
-        return self.copy_into_location(path,
-                                       file_format_type="CSV",
-                                       partition_by=partition_by,
-                                       overwrite=overwrite,
-                                       format_type_options=dict(compression=compression),
-                                       single=single)
+        Args:
+            location: The destination stage location.
+            partition_by: Specifies an expression used to partition the unloaded table rows into separate files. It can be a :class:`Column`, a column name, or a SQL expression.
+            format_type_options: Depending on the ``file_format_type`` specified, you can include more format specific options. Use the options documented in the `Format Type Options <https://docs.snowflake.com/en/sql-reference/sql/copy-into-location.html#format-type-options-formattypeoptions>`__.
+            header: Specifies whether to include the table column headings in the output files.
+            statement_params: Dictionary of statement level parameters to be set while executing this action.
+            copy_options: The kwargs that are used to specify the copy options. Use the options documented in the `Copy Options <https://docs.snowflake.com/en/sql-reference/sql/copy-into-location.html#copy-options-copyoptions>`__.
+            block: A bool value indicating whether this function will wait until the result is available.
+                When it is ``False``, this function executes the underlying queries of the dataframe
+                asynchronously and returns an :class:`AsyncJob`.
+        Returns:
+            A list of :class:`Row` objects containing unloading results.
+
+        Example::
+
+            >>> # save this dataframe to a csv file on the session stage
+            >>> df = session.create_dataframe([["John", "Berry"], ["Rick", "Berry"], ["Anthony", "Davis"]], schema = ["FIRST_NAME", "LAST_NAME"])
+            >>> remote_file_path = f"{session.get_session_stage()}/names.csv"
+            >>> copy_result = df.write.csv(remote_file_path, overwrite=True, single=True)
+            >>> copy_result[0].rows_unloaded
+            3
+        """
+
+        if len(copy_options) == 0:
+            return self.copy_into_location(
+                location,
```

Comment:
if `len(copy_options) == 0`, then `overwrite` argument will be empty and the default value when `overwrite` is not specified is already `FALSE`. see [linked](https://docs.snowflake.com/en/sql-reference/sql/copy-into-location#copy-options-copyoptions).


You don't need to add this special case, right? Same for below.